### PR TITLE
oauth-apiserver: add component e2e tests

### PR DESCRIPTION
**Pull request by stlaz**
/assign @sttts

### CODE REVIEW

File: ci-operator/config/openshift/oauth-apiserver/openshift-oauth-apiserver-master.yaml [hunk 1]
```diff
@@ -62,6 +62,14 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-e2e-aws
+- as: e2e-component
+  commands: |
+    if [ -d ./test/e2e ]; then
```
Comment (tnozicka) [inline ci-operator/config/openshift/oauth-apiserver/openshift-oauth-apiserver-master.yaml:67]: I do not fancy this approach.

This is always chicken&egg but rather then faking a result for e2e which clobbers history, we have conventionally delivered first the e2e code and as it isn't used it not critical to anything yet. Then setup a CI test check the rehearsals and iterate on the osin repo e2e code if needs be. Then when the e2e rehearsal is green, merge the release PR.

Comment (stlaz) [reply ci-operator/config/openshift/oauth-apiserver/openshift-oauth-apiserver-master.yaml]: fixed

**Comment by sttts** [comment]
/lgtm
/approve

/hold 

until rehearse job is green.

**Comment by stlaz** [comment]
/hold cancel
it's green, needs an approval from testplatform team

File: ci-operator/config/openshift/oauth-apiserver/openshift-oauth-apiserver-master.yaml [hunk 2]
```diff
@@ -62,6 +62,12 @@ tests:
   steps:
     cluster_profile: aws
     workflow: openshift-e2e-aws
+- as: e2e-component
+  commands: |
+    mkdir -p /tmp/artifacts
+    make test-e2e JUNITFILE=/tmp/artifacts/junit_report.xml --warn-undefined-variables
+  openshift_installer_src:
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/oauth-apiserver/openshift-oauth-apiserver-master.yaml:69]: Why not use the multi-stage workflow?

Comment (petr-muller) [reply ci-operator/config/openshift/oauth-apiserver/openshift-oauth-apiserver-master.yaml:69]: https://docs.ci.openshift.org/docs/how-tos/migrating-template-jobs-to-multistage/#openshift_installer_src

**Comment by stlaz** [comment]
/hold
lets' wait for the test to go green

**Pull request merged**